uber/models.py

```diff
@@ -728,21 +729,27 @@ def update_badge(self, attendee, old_badge_type, old_badge_num):
             from uber.badge_funcs import needs_badge_num
 
             if c.SHIFT_CUSTOM_BADGES and c.BEFORE_PRINTED_BADGE_DEADLINE:
-                # fill in the gap from the old number, if applicable
-                badge_num_keep = attendee.badge_num
-                if old_badge_num:
-                    self.shift_badges(old_badge_type, old_badge_num + 1, down=True)
```

Comment:
If `old_badge_type` and `attendee.badge_type` are the same, then we can constrain the range of badge numbers we have to shift. Instead of shifting the full range, we only need to shift those badges between `old_badge_num` and `attendee.badge_num`. This is more efficient, plus it means we only need to run a single badge shift query instead of the two we were running before.

The new implementation is a few lines below this.